src/dog.sol

```diff
@@ -176,6 +180,7 @@ contract Dog is LibNote {
             id = ClipperLike(milk.clip).kick({
                 tab: tab,
                 lot: dink,
+                lip: due / dink,
```

Comment:
I feel something is wrong with this math and the logic. Having a better collateralized CDP means a lower starting price. So let's say the feeds report a lower price than the real, a well collateralized CDP will be much more punished than one that's really underwater.
I feel the starting price should somehow be related to the inverse of the `due/dink`.

Reply:
There's no way for the auction contract to know, fundamentally, whether the feeds are wrong. The goal is to have some sort of fallback to a saner price if the feed price is super low. Even without this change, a very low oracle price punishes well-collateralized CDPs more (it's always worse to get liquidated if you have higher collateral/debt). Using liquidation price, corrected for `mat`, as a floor acts to make the worst-case less bad. Better to be liquidated at a low price than zero. I.e. this PR never makes the starting price any *lower* than it otherwise would, so it's still a strict improvement.

IMO it may be better to just invest in making the oracles more secure than trying to get extra clever here, although if you have any good ideas let's hear them :).

Reply:
I agree with @kmbarry1 that in an oracle attack scenario this outcome is strictly better than not having `lip`. I'm okay with leaving as is.

Reply:
Couldn't it happen that a very underwater vault won't be auctioned because the initial price is too high due this function?
I guess in that case it could be reset after the first cycle, which at least for now is not using this liquidation price.
Which btw, shouldn't we consider using it in the reset function somehow as well?

Reply:
Yes so this is something I wanted to bring up. I wasn't sure if we wanted to use it in the reset function, because then the auction's `top`s will never decrease during reset which would be bad in a scenario of a steep market decline. However if there was an oracle attack and no one bid on a CDP that was really expensive for example (highly collateralized), then on reset that price would go to zero.

Reply:
We could have another gov param that could be used to calculate the reset `top` using the previous `top`, decreasing it by a percentage.

Reply:
Hmmm you mean highly underwater, right? As if the CDP is highly collateralized, the liquidation will start at a lower price.
But yes, to your point, not really sure what to do with `reset`, as the same logic than in the originally liquidation would apply here, however the full property of `reset` wouldn't work if we do this.
I think these are the reasons why I feel this functionality is a bit forced.

Reply:
Yes sorry that's what I meant. And I do think that governance param solution could work for the reset, definitely open to other ideas though. I'll actually do a quick implementation of what I'm suggesting in the next commit.

Reply:
It might be interesting to do:
```
RESET_PRICE = max(buf*OSM_PRICE, PCT*top)
```
where `PCT` is the new gov parameter encoding the largest allowed (percentage-wise) decrease in `top`.

Reply:
Thats actually what I just did! In the last commit. 

`sales[id].top = max(rmul(sales[id].top, lop), rmul(rdiv(mul(uint256(val), BLN), spot.par()), buf));`

I made `lop` directly multiply instead of doing `sub(RAY, lop)` (E.g. 25% decrease => `lop = ray(0.75 ether)`). I did this because if `lop` isn't set for whatever reason it will just be zero, so it won't be considered in the equation. This is IMO better that it working out to be `1` if its not set, which will just set it back to the old `top`.

Reply:
Yep, that's what I had in mind.